### ISSUE TITLE
Fix out-of-date documentation on Formatter.java

### ISSUE THREAD

**Issue by jbduncan** [google/google-java-format#128]
In Formatter.java's [javadoc](https://github.com/google/google-java-format/blob/master/core/src/main/java/com/google/googlejavaformat/java/Formatter.java#L61), the following is said,

```
<p>This formatter uses the Eclipse parser to generate an AST...
```

I don't believe this is true any longer, as AFAICT google-java-format now uses OpenJDK's javac to do parsing.

**Comment by cushon** [comment]
Fixed in 878c113cc54d24da8aecbfbebdeb45f01748ec11.

**Comment by jbduncan** [comment]
Fantastic! Many thanks for fixing this @cushon. :+1:
